### PR TITLE
Fix inductor `sub` with symbolic integers.

### PR DESCRIPTION
**Pull request by ysiraichi**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #108518

Fix: #108159

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov

### CODE REVIEW

File: torch/_inductor/lowering.py [hunk 1]
```diff
@@ -377,12 +377,23 @@ def inner(*inputs: List[TensorBox], alpha=None):
                 inputs[-1] = mul(inputs[-1], alpha)
         else:
             assert alpha is None
+
+        # Get the first TensorBox/ExpandView input as the reference tensor for
+        # sizes, data types, and device.
```
Comment (peterbell10) [inline torch/_inductor/lowering.py:382]: `promote_constants` handles this for `ir.Constant` by wrapping it in an `ExpandView`:
https://github.com/pytorch/pytorch/blob/51c2e22e94caf56ac89967093b274ac81c7ca9e2/torch/_inductor/lowering.py#L351-L353

I think we could just do the same for `IndexingConstant` below:
https://github.com/pytorch/pytorch/blob/51c2e22e94caf56ac89967093b274ac81c7ca9e2/torch/_inductor/lowering.py#L356

File: torch/_inductor/lowering.py [hunk 2]
```diff
@@ -347,14 +347,12 @@ def const_func(x):
...
     for x in inputs:
-        if isinstance(x, (int, float)):
+        if isinstance(x, (int, float, sympy.Expr)):
             out.append(
                 ExpandView.create(
                     ir.Constant(x, ex.get_dtype(), ex.get_device()), list(ex.get_size())
                 )
             )
-        elif isinstance(x, sympy.Expr):
-            out.append(IndexingConstant(x, ex.get_dtype(), ex.get_device()))
```
Comment (peterbell10) [inline torch/_inductor/lowering.py:357]: You still need to create an `IndexingConstant` node here, just wrap it in `ExpandView.create`.

Comment (ysiraichi) [reply torch/_inductor/lowering.py:357]: Oh, I see. Thanks for the help, @peterbell10.

**Review by peterbell10 — APPROVED** [review]

**Pull request closed without merging**